zipline/data/data_portal.py

```diff
@@ -190,18 +187,21 @@ def handle_extra_source(self, source_df, sim_params):
         for group_name in group_names:
             group_dict[group_name] = grouped_by_sid.get_group(group_name)
 
+        extra_source_df = None
```

Comment:
yeah, that makes sense. Also, added a check in `get_fetcher_assets` to return `[]` when the df is empty